_posts/2016-11-09-Rust-1.13.md

```diff
@@ -0,0 +1,469 @@
+---
+layout: post
+title: "Announcing Rust 1.13.0"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.13.0. Rust is
+a systems programming language focused on safety, speed, and concurrency.
+
+As always, you can [install Rust 1.13.0][install] from the appropriate page on
+our website, and check out the [detailed release notes for 1.13.0][notes] on
+GitHub. 1448 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: 
+
+### What's in 1.13.0 stable
+
+The two big things in Rust 1.13 stable are compiler performance and a new
+operator, `?`. As usual, there's also a number of library stabilizations and
+other minor improvements and changes.
+
+#### The `?` operator
+
+Rust has gained a new operator, `?`, that's used to make error-handling more
+pleasant.
+
+Imagine we had some code to read some data from a file:
+
+```rust
+fn read_username_from_file() -> Result<String, io::Error> {
+    let f = File::open("username.txt");
+
+    let mut f = match f {
+        Ok(file) => file,
+        Err(e) => return Err(e),
+    };
+
+    let mut s = String::new();
+
+    match f.read_to_string(&mut s) {
+        Ok(_) => Ok(s),
+        Err(e) => Err(e),
+    }
+}
+```
+
+This code has two paths that can fail, opening the file and reading the
+data from it. If either of these fail to work, we'd like to return an
+error from `read_username_from_file`. Doing so involves `match`ing on
+the result of the I/O operations. While this gives us a lot of control
+over the details, in many cases, we just want to propogate any errors
+and write the "happy path" code.
+
+With `?`, this code looks like this:
+
+```rust
+fn read_username_from_file() -> Result<String, io::Error> {
+    let mut f = File::open("username.txt")?;
+    let mut s = String::new();
+
+    f.read_to_string(&mut s)?
+
+    Ok(s)
+}
+```
+
+In other words, `?` applies to a `Result` value, and if it was an `Ok`, it
+unwraps it and gives the inner value. If it was an `Err`, it returns from
+the function you're currently in.
+
+Seasoned Rustaceans may recongize that this is the same as the `try!` macro
```

Comment:
Can we do a quick search of crates.io to see how much `try!` is used?  I'd assume people are using it, but if it's not often used, perhaps it needs this introduction.